store/details_v2.go

```diff
@@ -349,6 +354,60 @@ func infoFromStoreSnap(d *storeSnap) (*snap.Info, error) {
 	return info, nil
 }
 
+func componentFromStoreResource(r storeResource) (*snap.Component, bool) {
+	compType, ok := strings.CutPrefix(r.Type, "component/")
+	if !ok {
+		return nil, false
+	}
+
+	comp := &snap.Component{
+		Name:        r.Name,
+		Summary:     r.Description.Clean(),
+		Description: r.Description.Clean(),
+		Type:        snap.ComponentType(compType),
+
+		// unable to fill the rest of the struct from a store resource
+	}
+
+	return comp, true
+}
+
+func addComponents(info *snap.Info, resources []storeResource) {
+	for _, r := range resources {
+		if comp, ok := componentFromStoreResource(r); ok {
+			if info.Components == nil {
+				info.Components = make(map[string]*snap.Component)
+			}
+
+			info.Components[comp.Name] = comp
+		}
+	}
+}
+
+func downloadInfoFromStoreDownload(d storeDownload) snap.DownloadInfo {
+	downloadInfo := snap.DownloadInfo{
+		DownloadURL: d.URL,
+		Size:        d.Size,
+		Sha3_384:    d.Sha3_384,
+	}
+
+	if len(d.Deltas) > 0 {
+		downloadInfo.Deltas = make([]snap.DeltaInfo, 0, len(d.Deltas))
+		for _, d := range d.Deltas {
+			downloadInfo.Deltas = append(downloadInfo.Deltas, snap.DeltaInfo{
```

Comment:
is this part ever reached if the call is made in the context of processing the download object of a resource?

Reply:
apparently there's no deltas for components, can you leave a comment about this in the code? I wonder if that's the case maybe we should stop processing early if the call is made in the context of a component